x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -230,27 +230,38 @@ protected final Page rows(List<TestCaseSupplier.TypedData> multirowFields) {
                 field -> assertThat("All multi-row fields must have the same number of rows", field.multiRowData(), hasSize(rowsCount))
             );
 
-        var blocks = new Block[multirowFields.size()];
+        List<Page> pages = new ArrayList<>();
 
-        for (int i = 0; i < multirowFields.size(); i++) {
-            var field = multirowFields.get(i);
-            try (
-                var wrapper = BlockUtils.wrapperFor(
-                    TestBlockFactory.getNonBreakingInstance(),
-                    PlannerUtils.toElementType(field.type()),
-                    rowsCount
-                )
-            ) {
+        int pageSize = randomIntBetween(1, 10);
```

Comment:
10,000 is a pretty normal size in production. Maybe `randomBoolean() ? between(1, 100) : Math.toIntExact(BlockFactory.DEFAULT_MAX_BLOCK_PRIMITIVE_ARRAY_SIZE.getBytes())`. Half the time we'll have degenerately small data and half the time it'll probably just be one page.